adapters/repos/db/index.go

```diff
@@ -180,9 +178,6 @@ func NewIndex(ctx context.Context, config IndexConfig,
 		metrics:         NewMetrics(logger, promMetrics, config.ClassName.String(), "n/a"),
 		centralJobQueue: jobQueueCh,
 	}
-	if class != nil && class.MultiTenancyConfig != nil {
-		index.tenantKey = class.MultiTenancyConfig.TenantKey
-	}
 
```

Comment:
We could make `PartitioningEnabled bool` a data member of `Index`.
This way we don't have to query the sharding state for a property which doesn't change once set